types/dummy/index.d.ts

```diff
@@ -3,3 +3,15 @@
  * ember-data. They will not be shipped to consumers. Ember-data still relies
  * on some private Ember APIs -- those should be defined here as we encounter them.
  */
+
+ declare module '@ember/array' {
+   export function detect(obj: any): boolean;
```

Comment:
this is probably best expressed as a user-defined type guard

Reply:
do you mean `unknown` instead of `any`? :confused:

Reply:
no, I mean
```ts
export function detect(obj: any): boolean is ArrayMixin;
```

Reply:
I think you mean this:

```ts
export function detect(obj: any): obj is ArrayMixin;
```

Reply:
but https://github.com/emberjs/ember.js/blob/9bdc0075762ac24bf18629d562548b140d0309cd/packages/%40ember/-internals/metal/lib/mixin.ts#L606

Reply:
`<item> is <Type>` is a specific kind of `boolean`. In this case, as I understand it, it's specifically detecting that this is an `ArrayMixin`, correct? If so, then you want the type guard form, not just `boolean`, because it lets TypeScript narrow the type so that when you call `.detect` on the other side it's confirmed to be an `ArrayMixin` from the type system's POV.

Reply:
See the [User-Defined Type Guards](https://www.typescriptlang.org/docs/handbook/advanced-types.html#user-defined-type-guards) section of the handbook for more.

Reply:
oh! I understand now, the `is` syntax is a like the `where` clause in C#.
Legit